packages/components/src/internal/components/samples/SampleDetailEditing.tsx

```diff
@@ -55,77 +84,113 @@ export class SampleDetailEditing extends PureComponent<Props, State> {
+            this.setState({ sampleTypeDomainFields, hasError: false });
+        } catch (e) {
+            this.setState({ hasError: true });
+        }
     };
 
-    canEdit(panelName: string): boolean {
-        const { canUpdate } = this.props;
-        if (canUpdate) return canUpdate(panelName);
-        return false;
-    }
+    getRow = (): Record<string, any> => {
+        const { queryGridModel, queryModel } = this.props;
 
-    getUpdateDisplayColumns = (isAliquot: boolean): any => {
-        const { sampleModel } = this.props;
+        if (queryModel) {
+            return queryModel.getRow();
+        } else {
+            return queryGridModel.getRow().toJS();
+        }
+    };
+
+    getUpdateDisplayColumns = (isAliquot: boolean): GroupedSampleDisplayColumns => {
+        const { queryGridModel, queryModel } = this.props;
         const { sampleTypeDomainFields } = this.state;
 
-        const allDisplayColumns = sampleModel.getDetailsDisplayColumns();
-        const allUpdateColumns = sampleModel.getUpdateDisplayColumns();
+        let displayColumns: List<QueryColumn>;
+        let updateColumns: List<QueryColumn>;
 
-        return getGroupedSampleDisplayColumns(allDisplayColumns, allUpdateColumns, sampleTypeDomainFields, isAliquot);
-    };
+        if (queryModel) {
+            displayColumns = List(queryModel.detailColumns);
+            updateColumns = List(queryModel.updateColumns);
+        } else {
+            displayColumns = queryGridModel.getDetailsDisplayColumns();
+            updateColumns = queryGridModel.getUpdateDisplayColumns();
+        }
 
-    renderAliquotDetailHeader = (row: any, aliquotHeaderDisplayColumns: List<QueryColumn>) => {
-        return <SampleAliquotDetailHeader row={row} aliquotHeaderDisplayColumns={aliquotHeaderDisplayColumns} />;
+        return getGroupedSampleDisplayColumns(displayColumns, updateColumns, sampleTypeDomainFields, isAliquot);
     };
 
     render() {
-        const { title, onUpdate, sampleModel, auditBehavior, detailRenderer } = this.props;
+        const {
+            actions,
+            auditBehavior,
+            canUpdate,
+            detailEditRenderer,
+            detailRenderer,
+            onEditToggle,
+            onUpdate,
+            queryGridModel,
+            queryModel,
+            title,
+        } = this.props;
         const { hasError, sampleTypeDomainFields } = this.state;
 
-        if (!sampleModel || !sampleModel.isLoaded) {
+        if (
+            !sampleTypeDomainFields ||
+            (queryModel && queryModel.isLoading) ||
+            (queryGridModel && !queryGridModel.isLoaded)
+        ) {
             return <LoadingPage title={title} />;
         }
 
-        const row = sampleModel.getRow();
-        const aliquotedFrom = row.getIn(['AliquotedFromLSID/Name', 'value']);
-
-        const isAliquot = aliquotedFrom != null;
-
-        if (hasError)
+        if (hasError) {
```

Comment:
It seems the error alert will never render because sampleTypeDomainFields will remain undefined in the case of error.

Reply:
Good catch. Fixed with e837fae.